src/pytorch_lightning/loggers/logger.py

```diff
@@ -355,10 +264,6 @@ def __getitem__(self, idx: int) -> "DummyLogger":
         # enables self.logger[0].experiment.add_image(...)
         return self
 
-    def __iter__(self) -> Generator[None, None, None]:
-        # if DummyLogger is substituting a logger collection, pretend it is empty
-        yield from ()
```

Comment:
This part is no longer necessary, was introduced here: https://github.com/Lightning-AI/lightning/pull/10232,